docs/dub-reference/dependencies.md

```diff
@@ -0,0 +1,25 @@
+# Dependency syntax
+
+## version
+### `"*"`
+### `~>a.b.c`
+### `~>a.b`
+### `>=a`, `>=a <b`, `<b`
+### `~gitbranch`
```

Comment:
@geod24 said not to document ~gitbranch, but I needed that to build `dlang-tour/english` which requires the git master version of `dlang-tour/core`. So if that isn't documented, I think we should document whatever the alternative is.

Reply:
the alternative is repository based dependencies, which is documented

See also: https://docs.webfreak.org/dub-reference/build_settings/#version-specifications

Reply:
OK, but I'm trying to replace the `dub run` command:

    dub run dlang-tour@~master -- --lang-dir .

I tried the following:
```
nick@nmt-pc:~/git/english$ dub run git+https://github.com/dlang-tour/core -- --lang-dir .
     Warning Package 'git+https://github.com/dlang-tour/core' was neither found locally nor online.
nick@nmt-pc:~/git/english$ dub run git+https://github.com/dlang-tour/core@master -- --lang-dir .
Error Invalid SemVer format: master
nick@nmt-pc:~/git/english$ dub run git+https://github.com/dlang-tour/core@~master -- --lang-dir .
Error (1): Error: Got JSON of type null_, expected object.
nick@nmt-pc:~/git/english$ dub run git+https://github.com/dlang-tour/core@6bc7780c48068ba2d87e99a27cfba64e689fb330 -- --lang-dir .
Error Invalid SemVer format: 6bc7780c48068ba2d87e99a27cfba64e689fb330
nick@nmt-pc:~/git/english$ dub run git+https://github.com/dlang-tour/core@~6bc7780c48068ba2d87e99a27cfba64e689fb330 -- --lang-dir .
Error (1): Error: Got JSON of type null_, expected object.
```

Reply:
Removed ~gitbranch and added a link

Reply:
ah yeah I don't think there is a shortcut format for that yet that would allow you to use it on the CLI like this. Might be worth a PR on dub itself

Reply:
OK thanks